modules/registry/registry.go

```diff
@@ -199,12 +198,60 @@ func (c *RegistryContainer) PushImage(ctx context.Context, ref string) error {
 
 	_, err = dockerCli.ImagePush(ctx, ref, pushOpts)
 	if err != nil {
-		return fmt.Errorf("failed to push image %s: %w", ref, err)
+		return fmt.Errorf("push image %q: %w", ref, err)
 	}
 
 	return c.ImageExists(ctx, ref)
 }
 
+// PullImage pulls an image from an external registry into the local Docker daemon.
+// Differently from PushImage, which uploads an image to the testcontainers managed registry,
+// this method downloads (copies) the specified image reference so it becomes
+// available locally for further operations such as tagging or pushing.
+func (c *RegistryContainer) PullImage(ctx context.Context, ref string) error {
+	dockerCli, err := testcontainers.NewDockerClientWithOpts(ctx)
+	if err != nil {
+		return fmt.Errorf("create docker client: %w", err)
+	}
+	defer dockerCli.Close()
+
+	pullOpts := image.PullOptions{
+		All:      false,
+		Platform: "linux/amd64",
```

Comment:
question: we are using `linux/amd64` because the underlying container is built in that platform, aren't we? 

For the image used in the tests, [`v.2.8.3`](https://hub.docker.com/layers/library/registry/2.8.3/images/sha256-46faa9a1ae6813194b53921a370f2f4f8c5e1aae228a89bceafef5847a6a3278), there exist multiple platforms, so if the underlying docker host is ARM, when pulling the registry image, the pull command will pull the arm-based image.

So my question is: would it make more sense to get the current platform in which the tests are run, so that it matched with the pulled registry image?

Reply:
LGTM, thanks